File: app/src/main/java/org/aossie/agoraandroid/utilities/SharedPrefs.java
package org.aossie.agoraandroid.utilities;

import android.content.Context;
import android.content.SharedPreferences;

@SuppressWarnings("SpellCheckingInspection")
public class SharedPrefs {

    private static final String myPrefs = "myPrefs";
    private static final String UserNameKey = "userName";
    private static final String EmailKey = "emailId";
    private static final String FullNameKey = "fullName";
    private static final String TokenKey = "token";
    private static final String PassKey = "userPass";
    private static final String ExpiresOnKey = "expiresOn";
    private final SharedPreferences sharedPreferences;
    private final SharedPreferences.Editor editor;


    public SharedPrefs(Context context) {
        sharedPreferences = context.getSharedPreferences(myPrefs, Context.MODE_PRIVATE);
        editor = sharedPreferences.edit();
    }


    //Save USerName in sharedPref and get Username
    public void saveUserName(String key) {
        editor.putString(UserNameKey, key);
        editor.commit();
    }

    public String getUserName() {
        return sharedPreferences.getString(UserNameKey, null);
    }


    //Save Email in sharedPref and get Email
    public void saveEmail(String email) {
        editor.putString(EmailKey, email);
        editor.commit();
    }

    public String getEmail() {
        return sharedPreferences.getString(EmailKey, null);
    }


    //Save firstName and lastName as fullName and get fullName
    public void saveFullName(String firstName, String lastName) {
        editor.putString(FullNameKey, firstName + " " + lastName);
        editor.commit();
    }

    public String getfullName() {
        return sharedPreferences.getString(FullNameKey, null);
    }

    //Save Token and get Token when required
    public void saveToken(String token) {
        editor.putString(TokenKey, token);
        editor.commit();
    }

    public String getToken() {
        return sharedPreferences.getString(TokenKey, null);
    }

    public void savePass(String password) {
        editor.putString(PassKey, password);
        editor.commit();
    }

    public String getPass() {
        return sharedPreferences.getString(PassKey, null);
    }

    public void saveTokenExpiresOn(String ExpiresOn) {
        editor.putString(ExpiresOnKey, ExpiresOn);
        editor.commit();
    }

    public String getTokenExpiresOn() {
        return sharedPreferences.getString(ExpiresOnKey, null);
    }

    public void clearLogin() {
        editor.putString(UserNameKey, null);
        editor.putString(EmailKey, null);
        editor.putString(FullNameKey, null);
        editor.commit();
    }
}

File: app/src/main/java/org/aossie/agoraandroid/utilities/SharedPrefs.kt
package org.aossie.agoraandroid.utilities

import android.content.Context
import android.content.SharedPreferences

class SharedPrefs(context: Context) {

  private val sharedPreferences: SharedPreferences
  private val editor: SharedPreferences.Editor

  companion object {
    private const val myPrefs = "myPrefs"
    private const val userNameKey = "userName"
    private const val firstNameKey = "firstName"
    private const val lastNameKey = "lastName"
    private const val emailKey = "emailId"
    private const val tokenKey = "token"
    private const val passKey = "userPass"
    private const val ExpiresOnKey = "expiresOn"
  }

  init {
    sharedPreferences = context.getSharedPreferences(myPrefs, Context.MODE_PRIVATE)
    editor = sharedPreferences.edit()
  }

  fun saveUserName(key: String?) {
    editor.putString(userNameKey, key)
    editor.commit()
  }

  val userName: String?
    get() = sharedPreferences.getString(userNameKey, null)

  fun saveEmail(email: String?) {
    editor.putString(emailKey, email)
    editor.commit()
  }

  val email: String?
    get() = sharedPreferences.getString(emailKey, null)

  fun saveFirstName(firstName: String?) {
    editor.putString(firstNameKey, firstName)
    editor.commit()
  }

  val firstName: String?
    get() = sharedPreferences.getString(firstNameKey, null)

  fun saveLastName(lastName: String?) {
    editor.putString(lastNameKey, lastName)
    editor.commit()
  }

  val lastName: String?
    get() = sharedPreferences.getString(lastNameKey, null)

  fun saveToken(token: String?) {
    editor.putString(tokenKey, token)
    editor.commit()
  }

  val token: String?
    get() = sharedPreferences.getString(tokenKey, null)

  fun savePass(password: String?) {
    editor.putString(passKey, password)
    editor.commit()
  }

  val pass: String?
    get() = sharedPreferences.getString(passKey, null)

  fun saveTokenExpiresOn(ExpiresOn: String?) {
    editor.putString(ExpiresOnKey, ExpiresOn)
    editor.commit()
  }

  val tokenExpiresOn: String?
    get() = sharedPreferences.getString(ExpiresOnKey, null)

  fun clearLogin() {
    editor.clear()
    editor.commit()
  }

}